packages/workspace/src/browser/workspace-service.ts

```diff
@@ -238,28 +274,34 @@ export class WorkspaceService implements FrontendApplicationContribution {
      * Adds a root folder to the workspace
      * @param uri URI of the root folder being added
      */
-    async addRoot(uri: URI): Promise<void> {
+    async addRoot(uri: URI, metaDataFile: URI | undefined): Promise<void> {
```

Comment:
why do we need to pass a URI to workspace file here? Should we just add it to the current one and make sure that it is set before?